src/services/analytics/TagManager.ts

```diff
@@ -1,94 +1,89 @@
-// Based on https://github.com/alinemorelli/react-gtm
+import Cookies from 'js-cookie'
+
+import { IS_PRODUCTION } from '@/config/constants'
 
 type DataLayer = Record<string, unknown>
 
 export type TagManagerArgs = {
-  /**
-   * GTM id, must be something like GTM-000000.
-   */
+  // GTM id, e.g. GTM-000000
   gtmId: string
-  /**
-   * Used to set environments.
-   */
-  auth?: string | undefined
-  /**
-   * Used to set environments, something like env-00.
-   */
-  preview?: string | undefined
-  /**
-   * Object that contains all of the information that you want to pass to Google Tag Manager.
-   */
-  dataLayer?: DataLayer | undefined
-}
-
-export const DATA_LAYER_NAME = 'dataLayer'
-
-export const _getRequiredGtmArgs = ({ gtmId, dataLayer = undefined, auth = '', preview = '' }: TagManagerArgs) => {
-  return {
-    gtmId,
-    dataLayer,
-    auth: auth ? `&gtm_auth=${auth}` : '',
-    preview: preview ? `&gtm_preview=${preview}` : '',
-  }
+  // GTM authetication key
+  auth: string
+  // GTM environment, e.g. env-00.
+  preview: string
+  // Object that contains all of the information that you want to pass to GTM
+  dataLayer?: DataLayer
 }
 
-// Initialization scripts
-
-export const _getGtmScript = (args: TagManagerArgs) => {
-  const { gtmId, auth, preview } = _getRequiredGtmArgs(args)
+const DATA_LAYER_NAME = 'dataLayer'
 
-  const script = document.createElement('script')
+const TagManager = {
+  // `jest.spyOn` is not possible if outside of `TagManager`
+  _getScript: ({ gtmId, auth, preview }: TagManagerArgs) => {
+    const script = document.createElement('script')
 
-  const gtmScript = `
+    const gtmScript = `
     (function (w, d, s, l, i) {
       w[l] = w[l] || [];
       w[l].push({ 'gtm.start': new Date().getTime(), event: 'gtm.js' });
       var f = d.getElementsByTagName(s)[0],
-        j = d.createElement(s),
+      j = d.createElement(s),
         dl = l != 'dataLayer' ? '&l=' + l : '';
-      j.async = true;
-      j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl + '${auth}${preview}&gtm_cookies_win=x';
-      f.parentNode.insertBefore(j, f);
-    })(window, document, 'script', '${DATA_LAYER_NAME}', '${gtmId}');`
+        j.async = true;
+        j.src = 'https://www.googletagmanager.com/gtm.js?id=' + i + dl + '&gtm_auth=${auth}&gtm_preview=${preview}&gtm_cookies_win=x';
+        f.parentNode.insertBefore(j, f);
+      })(window, document, 'script', '${DATA_LAYER_NAME}', '${gtmId}');`
 
-  script.innerHTML = gtmScript
+    script.innerHTML = gtmScript
 
-  return script
-}
-
-// Data layer scripts
+    return script
+  },
+  isInitialized: () => {
+    const GTM_SCRIPT = 'https://www.googletagmanager.com/gtm.js'
 
-export const _getGtmDataLayerScript = (dataLayer: DataLayer) => {
-  const script = document.createElement('script')
+    return !!document.querySelector(`[src^="${GTM_SCRIPT}"]`)
```

Comment:
Would it work if we checked that `window.dataLayer` exists instead? It would be a bit simpler than querying the DOM.

Reply:
I would sooner check the DOM. We could manually push to `window.dataLayer` and then this would return `true` when GTM isn't initialised.